src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_ecc_import_export.c

```diff
@@ -0,0 +1,641 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_ecc_import_export.h"
+#include "pal_bignum.h"
+#include "pal_eckey.h"
+#include "pal_jni.h"
+#include "pal_utilities.h"
+
+static jobject CryptoNative_CreateBigInteger(uint8_t* data, int32_t len)
+{
+    JNIEnv* env = GetJNIEnv();
+    jbyteArray buffArray = (*env)->NewByteArray(env, len);
+    (*env)->SetByteArrayRegion(env, buffArray, 0, len, (jbyte*)data);
+    jobject bigNum;
+    if (data[0] & 0x80)
+    {
+        // If the top bit is set, we need to create a BigInteger with the sign-and-magnitude constructor.
+        bigNum = (*env)->NewObject(env, g_bigNumClass, g_bigNumCtorWithSign, 1, buffArray);
+    }
+    else
+    {
+        // Otherwise, we need to create it with the byte-array-value constructor.
+        bigNum = (*env)->NewObject(env, g_bigNumClass, g_bigNumCtor, buffArray);
+    }
+    (*env)->DeleteLocalRef(env, buffArray);
+    return CheckJNIExceptions(env) ? FAIL : ToGRef(env, bigNum);
+}
+
+int32_t CryptoNative_GetECKeyParameters(const EC_KEY* key,
+                                        int32_t includePrivate,
+                                        jobject* qx,
+                                        int32_t* cbQx,
+                                        jobject* qy,
+                                        int32_t* cbQy,
+                                        jobject* d,
+                                        int32_t* cbD)
+{
+    assert(qx != NULL);
+    assert(cbQx != NULL);
+    assert(qy != NULL);
+    assert(cbQy != NULL);
+    assert(d != NULL);
+    assert(cbD != NULL);
+
+    JNIEnv* env = GetJNIEnv();
+
+    // Get the public key
+    jobject publicKey = (*env)->CallObjectMethod(env, key->keyPair, g_keyPairGetPublicMethod);
```

Comment:
I feel like there are a lot more `ON_EXCEPTION_GOTO(fail)`s (or whatever the right words are :smile:) needed, since anything that allocates a return could OOM.

Reply:
We've decided that since OOM is already a fatal exception, once we hit OOM we're probably going down anyway. As a result, we don't need to check exceptions after every call since JNI will take down the process if there's an exception in progress when we call another JNI method.